src/NuGet.Clients/NuGet.CommandLine/Commands/ListCommand.cs

```diff
@@ -1,91 +1,25 @@
+        "ListCommandDescription",
+        UsageSummaryResourceName = "ListCommandUsageSummary",
+        UsageDescriptionResourceName = "ListCommandUsageDescription",
+        UsageExampleResourceName = "ListCommandUsageExamples")]
+    [Obsolete(message:"Use SearchCommand class. This class will disappear in upcoming releases", error: false)]
+    public class ListCommand : SearchCommand
     {
-        private readonly List<string> _sources = new List<string>();
-
-        [Option(typeof(NuGetCommand), "ListCommandSourceDescription")]
-        public ICollection<string> Source
-        {
-            get { return _sources; }
-        }
-
-        [Option(typeof(NuGetCommand), "ListCommandVerboseListDescription")]
-        public bool Verbose { get; set; }
-
-        [Option(typeof(NuGetCommand), "ListCommandAllVersionsDescription")]
-        public bool AllVersions { get; set; }
-
-        [Option(typeof(NuGetCommand), "ListCommandPrerelease")]
-        public bool Prerelease { get; set; }
-
-        [Option(typeof(NuGetCommand), "ListCommandIncludeDelisted")]
-        public bool IncludeDelisted { get; set; }
-
-        private IList<Configuration.PackageSource> GetEndpointsAsync()
-        {
-            var configurationSources = SourceProvider.LoadPackageSources()
-                .Where(p => p.IsEnabled)
-                .ToList();
-
-            IList<Configuration.PackageSource> packageSources;
-            if (Source.Count > 0)
-            {
-                packageSources = Source
-                    .Select(s => Common.PackageSourceProviderExtensions.ResolveSource(configurationSources, s))
-                    .ToList();
-            }
-            else
-            {
-                packageSources = configurationSources;
-            }
-            return packageSources;
-        }
-
         public async override Task ExecuteCommandAsync()
         {
-            if (Verbose)
-            {
-                Console.WriteWarning(LocalizedResourceManager.GetString("Option_VerboseDeprecated"));
-                Verbosity = Verbosity.Detailed;
-            }
-
-            var listCommandRunner = new ListCommandRunner();
-            var listEndpoints = GetEndpointsAsync();
-
-            var list = new ListArgs(Arguments,
-                listEndpoints,
-                Settings,
-                Console,
-                Console.PrintJustified,
-                Verbosity == Verbosity.Detailed,
-                LocalizedResourceManager.GetString("ListCommandNoPackages"), 
-                LocalizedResourceManager.GetString("ListCommand_LicenseUrl"),
-                LocalizedResourceManager.GetString("ListCommand_ListNotSupported"),
-                AllVersions, 
-                IncludeDelisted,
-                Prerelease,
-                CancellationToken.None);
-
-            await listCommandRunner.ExecuteCommand(list);
+            Console.WriteWarning(LocalizedResourceManager.GetString("ListCommandDeprecatedMessage"));
```

Comment:
Why remove the complete functionality? 

I was under the impression that we want to preserve the behavior with a warning message.
 
This way, I don't see it adding much value. 
If someone already  has a script that runs `nuget.exe list [some valid params]`, then it would suddenly start failing. 

//cc @anangaur

Reply:
@dominoFire 
Wait for @rrelyea/ @anangaur before making changes.

Reply:
As far as I can tell, it's not removing the functionality. The class now extends `SearchCommand` instead of `Command`, and the line below the line you selected for this comment calls `base.ExecuteCommandAsync`, hence the existing list functionality remains.

Reply:
I didn't see that, totally my bad. 
If the behavior is preserved, then you can resolve this comment @dominoFire

Reply:
There's no change in functionality. I only renamed ListCommand to SearchCommand and created a new ListCommand class that calls the functionality of SearchCommand.